manim/utils/space_ops.py

```diff
@@ -802,14 +802,16 @@ def earclip_triangulation(verts: np.ndarray, ring_ends: list) -> list:
     return [indices[mi] for mi in meta_indices]
 
 
-def cartesian_to_spherical(vec: Sequence[float]) -> np.ndarray:
+def cartesian_to_spherical(
+    vec: np.ndarray | Sequence[float],
+) -> np.ndarray | Sequence[float]:
```

Comment:
The new `Vector3DLike` type alias from #4068 seems fit for this task.

In order to make this function more consistent by always returning a NumPy array (`Vector3D`), can you change the code inside the `if norm == 0:` branch so that it returns `np.zeros(3)` instead of `(0, 0, 0)`?
```suggestion
def cartesian_to_spherical(vec: Vector3DLike) -> Vector3D:
```